### PR TITLE
Add contribution guide for CEXes and for Ethereum modules.

### LINKED ISSUE #3488: Add contribution guide for CEX and DeFi modules
## Problem Definition

There is no easy way for external people to see how to create rotki modules. We need to change that. Not all parts of the code are super modular but for those that are there is no way other than reading the code to understand how to contribute

## Task

- [x] Add a guide in the contribution docs about how can a user add support for a new CEX.
- [x] Add a guide in the contributions docs about how can a user add support for a new DeFi module

Note for future: Iterating through these guides we should also find ways to make the tool more modular.

### PR DESCRIPTION
**Pull request by LefterisJP**
Closes #3488 

- Add contribution guide for CEXes and for Ethereum modules.
- Fix some broken parts in the docs

### CODE REVIEW

File: docs/contribute.rst [hunk 1]
```diff
@@ -156,6 +156,167 @@ Helpful commands
...
+
+
+
+Create exchange module
+--------------------------
+
+To add a new CEX you should create a new file with the name of the exchange all lowercase in `here <https://github.com/rotki/rotki/tree/develop/rotkehlchen/exchanges>`__.
+
+It should have a class which should be the exact same name as the file but with the first letter capitalized.
+
```
Comment (yabirgb) [inline docs/contribute.rst:181]: Maybe provide an example here?

File: docs/contribute.rst [hunk 2]
```diff
@@ -156,6 +156,167 @@ Helpful commands
...
+These are defined in the ``constants.py`` file.
+
+Mappings and rules
+^^^^^^^^^^^^^^^^^^^
+
+The ``addresses_to_decoders()`` method maps any contract addresses that are identified in the transaction with the specific decoding function that can decode it. This is optional.
+
+The ``decoding_rules()`` defines any functions that should simply be used for all decoding so long as this module is active. This is optional.
+
+The ``enricher_rules()`` defies any functions that woudl be used for as long as this module is active to analyze already existing decoded events and enrich them with extra information we can decode thanks to this module. This is optional.
```
Comment (yabirgb) [inline docs/contribute.rst:256]:
```suggestion
The ``enricher_rules()`` defies any functions that would be used for as long as this module is active to analyze already existing decoded events and enrich them with extra information we can decode thanks to this module. This is optional.
```

File: docs/contribute.rst (hunk 1 above)
Comment (LefterisJP) [reply docs/contribute.rst:181]: I did. I linked to one.

File: docs/contribute.rst (hunk 2 above)
Comment (LefterisJP) [reply docs/contribute.rst:256]: argh thanks.

File: docs/contribute.rst [hunk 3]
```diff
@@ -396,4 +557,4 @@ tag name.
     docker run --rm --privileged tonistiigi/binfmt:latest --install arm64
     docker buildx create --name imgbldr --use
     docker buildx inspect --bootstrap --builder imgbldr
-    docker buildx build --build-arg REVISION='git sha' --build-arg ROTKI_VERSION=vx.x.x --file ./Dockerfile --platform linux/amd64 --platform linux/arm64 --tag rotki/rotki:vx.x.x --tag rotki/rotki:latest --push .
\ No newline at end of file
+    docker buildx build --build-arg REVISION='git sha' --build-arg ROTKI_VERSION=vx.x.x --file ./Dockerfile --platform linux/amd64 --platform linux/arm64 --tag rotki/rotki:vx.x.x --tag rotki/rotki:latest --push .
```
Comment (yabirgb) [inline docs/contribute.rst]: Why is this line shown in the diff?

**Review by yabirgb — APPROVED** [review]
Thanks <3

File: docs/contribute.rst (hunk 3 above)
Comment (LefterisJP) [reply docs/contribute.rst]: was probably missing or not missing newline and emacs fixed it.

**Pull request merged**